File: templategenerator.systemtest/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/util/ProcessExecutor.java
package com.optivem.atddaccelerator.templategenerator.systemtest.util;

import java.io.BufferedReader;
import java.io.InputStreamReader;

import static org.assertj.core.api.Assertions.assertThat;

public class ProcessExecutor {

    public static ProcessResult executeProcess(String... command)
    {
        var processBuilder = new ProcessBuilder(command);

        int exitCode;
        StringBuilder output = new StringBuilder();
        StringBuilder errors = new StringBuilder();

        try {
            Process process = processBuilder.start();

            // Read stdout
            BufferedReader stdout = new BufferedReader(new InputStreamReader(process.getInputStream()));

            String line;
            while ((line = stdout.readLine()) != null) {
                output.append(line).append("\n");
            }

            // Read stderr
            BufferedReader stderr = new BufferedReader(new InputStreamReader(process.getErrorStream()));

            while ((line = stderr.readLine()) != null) {
                errors.append(line).append("\n");
            }

            exitCode = process.waitFor();

        }catch(Exception e) {
            throw new RuntimeException("Failed to execute script", e);
        }

        var outputStr = output.toString();
        var errorsStr = errors.toString();

        return new ProcessResult(exitCode, outputStr, errorsStr);
    }
}

File: tests/Util/Process/ProcessExecutor.cs
using System.Diagnostics;
using System.Text;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process
{
    public static class ProcessExecutor
    {
        public static ProcessResult ExecuteProcess(params string[] command)
        {
            if (command.Length == 0)
                throw new ArgumentException("Command cannot be empty", nameof(command));

            var processInfo = new ProcessStartInfo
            {
                FileName = command[0],
                Arguments = string.Join(" ", command.Skip(1).Select(arg => $"\"{arg}\"")),
                RedirectStandardOutput = true,
                RedirectStandardError = true,
                UseShellExecute = false,
                CreateNoWindow = true
            };

            var output = new StringBuilder();
            var errors = new StringBuilder();

            try
            {
                using var process = new System.Diagnostics.Process { StartInfo = processInfo };
                
                process.OutputDataReceived += (sender, e) =>
                {
                    if (e.Data != null)
                        output.AppendLine(e.Data);
                };

                process.ErrorDataReceived += (sender, e) =>
                {
                    if (e.Data != null)
                        errors.AppendLine(e.Data);
                };

                process.Start();
                process.BeginOutputReadLine();
                process.BeginErrorReadLine();
                process.WaitForExit();

                return new ProcessResult(process.ExitCode, output.ToString(), errors.ToString());
            }
            catch (Exception e)
            {
                throw new InvalidOperationException("Failed to execute script", e);
            }
        }

        public static void WaitTime(int millis)
        {
            Thread.Sleep(millis);
        }
    }
}